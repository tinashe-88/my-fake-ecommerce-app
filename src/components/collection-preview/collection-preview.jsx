import React from 'react'

import CollectionItem from '../collection-item/collection-item'

import {
  CollectionPreviewContainer,
  TitleContainer,
  PreviewContainer
} from './collection-preview.styles'

const CollectionPreview = ({ title, items }) => (
  <CollectionPreviewContainer>
    <TitleContainer>
      {title.toUpperCase()}
    </TitleContainer>
    <PreviewContainer>
      {
        items
          .filter((item, i) => i < 4)
          .map(item => (
            <CollectionItem key={item.id} item={item}/>
        ))
      }
    </PreviewContainer>
  </CollectionPreviewContainer>
);

export default CollectionPreview