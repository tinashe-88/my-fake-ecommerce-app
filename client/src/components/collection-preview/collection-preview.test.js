import React from 'react'
import { shallow } from 'enzyme'

import { CollectionPreview } from './collection-preview'

describe('CollectionPreview component', () => {
  let wrapper
  let mockMatch
  let mockHistory
  const mockRouteName = 'jackets'

  beforeEach(() => {
    mockMatch = {
      path: '/shop'
    }

    mockHistory = {
      push: jest.fn()
    }

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      routeName: mockRouteName,
      title: 'jackets',
      items: []
    }

    wrapper = shallow(<CollectionPreview {...mockProps}/>)
  })

  it('Should render CollectionPreview component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call history,push when the correct string in TitleContainer is clicked', () => {
    wrapper.find('TitleContainer').simulate('click')
    expect(mockHistory.push).toHaveBeenCalledWith(
      `${mockMatch.path}/${mockRouteName}`
    )
  })
})