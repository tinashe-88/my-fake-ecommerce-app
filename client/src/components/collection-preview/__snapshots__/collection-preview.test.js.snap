// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionPreview component Should render CollectionPreview component 1`] = `
<styled.div>
  <TitleContainer
    onClick={[Function]}
  >
    JACKETS
  </TitleContainer>
  <styled.div />
</styled.div>
`;
